test/DynamoCoreWpfTests/RecordedTests.cs

```diff
@@ -3827,6 +3829,1109 @@ public void Lacing_Deffect_5759()
+                    AssertPreviewCount("016b4fbe-5a97-4308-9ab6-2950d3c36f1e", 20);//check Math.DegreesToRadians
+                    Assert.AreEqual(GetPreviewValue("f96439d0-6f79-4a30-955c-80e826a6ab69").ToString(), "Point(X = 10.000, Y = 20.000, Z = 0.000)");               
+                }
+            });
+        }
+
+
+        [Test]
+        public void MAGN_7348_Math_Point()
+        {
+            // Check Math function
+            // Scenario
+            //  a) By connecting node
+            //  b) compare two lines' values 
+            // http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7348
+            preloadGeometry = true;
+            RunCommandsFromFile("MAGN_7348_Math_Point.xml", (commandTag) =>
+            {
+                var workspace = ViewModel.Model.CurrentWorkspace;
+               
+                if (commandTag == "FirstRun")
+                {
+                    var line = GetPreviewValue("e615f564-b830-48cd-adfa-06c846b6c91b");
+                    Assert.IsNotNull(line);
+                    AssertPreviewValue("5a28bff1-af6f-4236-b924-ff43a4e5efb4", 247.821);
+                    var line2 = GetPreviewValue("cfb83855-b371-4ceb-b816-31109afd708f");
+                    Assert.IsNotNull(line2);
+                    Assert.AreEqual(line.ToString(), line2.ToString());                  
+                }                           
+            });
+        }
+
+
+        [Test]
+        public void MAGN_7348_Math_Point_Formula()
+        {
+            // Check Math function
+            // Scenario
+            //  a) By connecting node
+            //  b) change point lacing from cross product to shortest 
+            // http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7348
+            preloadGeometry = true;
+            RunCommandsFromFile("MAGN_7348_Math_Point_Formula.xml", (commandTag) =>
+            {
+                var workspace = ViewModel.Model.CurrentWorkspace;
+               
+                if (commandTag == "FirstRun")
+                {
+                    AssertPreviewCount("eb39e7be-b421-4083-9e50-b628a78fe02e", 20);//check nurbscurve
+                    NodeModel node = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace
+                       ("54648e79-9127-445c-a3c9-7c59b0453880");// check surface.byloft
+                    Assert.AreEqual(ElementState.Active, node.State);
+                    Assert.AreEqual(GetPreviewValue("54648e79-9127-445c-a3c9-7c59b0453880").ToString(),"Surface");
+                }
+                else if (commandTag == "SecondRun")
+                {
+                    NodeModel node = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace
+                      ("54648e79-9127-445c-a3c9-7c59b0453880");// check surface.byloft
+                    Assert.AreEqual(ElementState.Warning, node.State);
+                }     
+            });
+        }
+
+
+        [Test, Category("Failure")]
```

Comment:
WriteToExcel should not give any warning, warning is "null value cannot be cascaded to int32".
But it should give a list of value.